src/mailchimp/mailchimp_manager.py

```diff
@@ -85,8 +115,16 @@ def create_folder(self, foldername: str) -> int:
         Raises:
             TODO: Document all possible errors
         """
-        raise NotImplementedError
-
+        if type(foldername) != str:
+            raise ValueError
+        elif foldername = "":
+            raise ValueError
+        elif len(foldername) <= 4:
+            raise ValueError
```

Comment:
because i accounted for the extension being 4 characters so ".txt" itself is 4 characters hence the foldername has to be greater than 4 characters to be valid